packages/treasury/src/liquidateMinimum.js

```diff
@@ -51,21 +45,17 @@ export async function start(zcf) {
         debtorSeat,
       );
 
-      // if swapOut failed for insufficient funds, we'll sell it all
-      async function onSwapOutFail(error) {
-        const strategy = makeDefaultLiquidationStrategy(autoswap);
+      // if swapOut failed to make the trade, we'll sell it all
+      async function onSwapOutNoTrade() {
+        const strategy = makeDefaultLiquidationStrategy(amm);
         trace(`onSwapoutFail`);
-        assert(
-          error.message.match(AutoswapInsufficientMsg),
```

Comment:
good riddance to that one. errors are best treated as sealed to normal callers.

Reply:
That might not be a mistake, but there's a slippery slope to peeking at "this reason" and getting stuff that you shouldn't; it was enough of a risk that Joe-E ruled out catching exceptions altogether.

@erights I'm not sure we have a good answer to this question documented. The closest I can find is:

> A pervasive concern is hiding diagnostic information—both for confidentiality and for deterministic replay. Code that obtains access to an error object, for example by catching it, should not have access this hidden diagnostic information. -- https://github.com/endojs/endo/tree/master/packages/ses/src/error#hiding-and-revealing-local-diagnostic-information

The preferred idiom in E was ejectors: if you think something might fail, you pass in an ejector that the callee can use to signal that yup, it failed for one of the expected reasons. The callee can pass args to the ejector in the clear. Adding those to agoric-sdk / endo is on the todo list https://github.com/endojs/endo/issues/414 -> 
 https://github.com/Agoric/SES/blob/0.5.0/src/old/ejectorsGuardsTrademarks.js